src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -300,10 +300,41 @@ internal unsafe struct MethodTable
         private uint Flags;
         [FieldOffset(4)]
         public uint BaseSize;
+        [FieldOffset(0x0e)]
+        public ushort InterfaceCount;
+        [FieldOffset(ParentMethodTableOffset)]
+        // NOTE: the ParentMethodTable could be an indirect pointer. (see: enum_flag_HasIndirectParent).
```

Comment:
It can also be a relative pointer. All these tweaks apply for fragile NGen only that is disabled by default. I do not think we need to worry about fragile NGen support, or adding comments like this. It will be deleted at some point.
 
If you would like keep it working for now, you can always use the Any version of the helpers for fragile NGen - just like what you have done for interfaces.

Reply:
I will remove the comment. I was not sure if that is supported or not. 
It did gave me a pause to think how we should deal with indirection.

It seems we could follow the parent chain naturally here though, even if indirect case was possible. 

---
In reply to: [368312225](https://github.com/dotnet/runtime/pull/1068#discussion_r368312225) [](ancestors = 368312225)

Reply:
I will remove the comment. I was not sure if that is supported or not. 
It did gave me a pause to think how we should deal with indirection.

It seems we could follow the parent chain naturally here though, even if indirect case was possible. 

---
In reply to: [368312225](https://github.com/dotnet/runtime/pull/1068#discussion_r368312225) [](ancestors = 368312225)